circulating_supply/src/main.rs

```diff
@@ -2,39 +2,54 @@ use std::ops::{Add, Sub};
 use std::str::FromStr;
 use std::time::{Duration, SystemTime};
 
-use actix::Addr;
-use actix_diesel::Database;
 use anyhow::Context;
 use bigdecimal::{BigDecimal, ToPrimitive};
 use chrono::NaiveDateTime;
-use diesel::PgConnection;
 use tracing::{error, info, warn};
 
-use near_indexer::near_primitives;
+use near_jsonrpc_client::{methods, JsonRpcClient};
 
-use crate::aggregated::{account_details, circulating_supply};
-use crate::db_adapters::accounts;
-use crate::db_adapters::aggregated::circulating_supply::{
-    add_circulating_supply, get_precomputed_circulating_supply_for_timestamp,
-};
-use crate::db_adapters::blocks;
-use crate::models;
-use crate::models::aggregated::circulating_supply::CirculatingSupply;
+use explorer_database::{adapters, models};
 
+mod account_details;
 mod lockup;
 mod lockup_types;
 
 const DAY: Duration = Duration::from_secs(60 * 60 * 24);
 const RETRY_DURATION: Duration = Duration::from_secs(60 * 60 * 2);
 
+const CIRCULATING_SUPPLY: &str = "circulating_supply";
+
+#[actix::main]
+async fn main() {
+    dotenv::dotenv().ok();
+
+    tracing_subscriber::fmt::Subscriber::builder()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .init();
+
+    let pool = models::establish_connection(
+        &std::env::var("DATABASE_URL")
+            .expect("DATABASE_URL must be set in either .env or environment "),
+    );
+
+    let rpc_client = JsonRpcClient::connect(
+        std::env::var("RPC_URL").expect("RPC_URL must be set in either .env or environment"),
+    );
+
+    info!(target: crate::CIRCULATING_SUPPLY, "Starting calculations");
+
+    run_circulating_supply_computation(rpc_client, pool).await;
```

Comment:
Intentionally haven't included the "is mainnet" check here as we'll _only_ run this for mainnet.